core/src/main/java/org/apache/carbondata/core/dictionary/generator/key/DictionaryMessage.java

```diff
@@ -55,6 +60,10 @@ public void readData(ByteBuf byteBuf) {
     byteBuf.readBytes(tableBytes);
     tableUniqueName = new String(tableBytes);
 
+    byte[] tableIdBytes = new byte[byteBuf.readInt()];
+    byteBuf.readBytes(tableIdBytes);
+    tableUniqueId = new String(tableIdBytes);
```

Comment:
creating a string object for every message is a costly operation. Read bytes and return the byte array and maintain a map with byte array as key object